### PR TITLE
[CI] Add some notation for train_full.py and move it to a new directory

### PR DESCRIPTION
**Pull request by caojy1998**
## Description

1 Add highlights and flowchart for train_full.py.
2 Move train_full.py to examples/core/graphsage/node_classification.py.
3 Change some variable name in the forward function to make it more understandable.
4 Change the sting in print function to fstring.
5 Delete the dropout layer for input in the forward function.

## Checklist
Please feel free to remove inapplicable items for your PR.
- [ ] The PR title starts with [$CATEGORY] (such as [NN], [Model], [Doc], [Feature]])
- [ ] I've leverage the [tools](https://docs.google.com/document/d/1iHyj7zlmygKSk5gBPsqIqL5ASPzJSPREaNT_QdsiYA4/edit) to beautify the python and c++ code.
- [ ] The PR is complete and small, read the [Google eng practice (CL equals to PR)](https://google.github.io/eng-practices/review/developer/small-cls.html) to understand more about small PR. In DGL, we consider PRs with less than 200 lines of core code change are small (example, test and documentation could be exempted).
- [ ] All changes have test coverage
- [ ] Code is well-documented
- [ ] To the best of my knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change
- [ ] Related issue is referred in this PR
- [ ] If the PR is for a new model/paper, I've updated the example index [here](../examples/README.md).

### CODE REVIEW

**Comment by frozenbugs** [comment]
Install updating the stale code, can you copy this to examples/core/graphsage/node_classification.py?

File: examples/core/graphsage/node_classification.py [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+from dgl.data import CiteseerGraphDataset, CoraGraphDataset, PubmedGraphDataset
+
+
+class SAGE(nn.Module):
+    def __init__(self, in_size, hid_size, out_size):
+        super().__init__()
+        self.layers = nn.ModuleList()
+        # two-layer GraphSAGE-mean
+        self.layers.append(dglnn.SAGEConv(in_size, hid_size, "gcn"))
+        self.layers.append(dglnn.SAGEConv(hid_size, out_size, "gcn"))
```
Comment (frozenbugs) [inline examples/core/graphsage/node_classification.py:38]: The aggregator type says gcn, while the comment says GraphSAGE-mean, is it correct?

File: examples/core/graphsage/node_classification.py [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+import torch.nn.functional as F
+from dgl import AddSelfLoop
+from dgl.data import CiteseerGraphDataset, CoraGraphDataset, PubmedGraphDataset
+
+
+class SAGE(nn.Module):
+    def __init__(self, in_size, hid_size, out_size):
+        super().__init__()
+        self.layers = nn.ModuleList()
+        # two-layer GraphSAGE-mean
```
Comment (frozenbugs) [inline examples/core/graphsage/node_classification.py:36]: For comment, capitalize the first char and end with dot.
```suggestion
        # Two-layer GraphSAGE-mean.
```

File: examples/core/graphsage/node_classification.py [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+        self.dropout = nn.Dropout(0.5)
+
+    def forward(self, graph, x):
+        h = self.dropout(x)
+        for l, layer in enumerate(self.layers):
+            h = layer(graph, h)
+            if l != len(self.layers) - 1:
+                h = F.relu(h)
+                h = self.dropout(h)
+        return h
```
Comment (frozenbugs) [inline examples/core/graphsage/node_classification.py:48]: Can we improve readability here?
essentially it does:
- 1. dropout
- 1. layer
- 1. relu
- 2. dropout
- 2. layer
- 2. relu
(is_last_layer)
- 3. dropout
- 3. layer

File: examples/core/graphsage/node_classification.py [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+    model = SAGE(in_size, 16, out_size).to(device)
+
+    # model training
+    print("Training...")
+    train(g, features, labels, masks, model)
+
+    # test the model
+    print("Testing...")
+    acc = evaluate(g, features, labels, g.ndata["test_mask"], model)
+    print("Test accuracy {:.4f}".format(acc))
```
Comment (keli-wen) [inline examples/core/graphsage/node_classification.py:132]: We'd better use fstring.

Comment (keli-wen) [reply examples/core/graphsage/node_classification.py:132]: You can check the whole example about what can be changed to fstring.

File: examples/core/graphsage/node_classification.py [hunk 5]
```diff
@@ -0,0 +1,132 @@
...
+
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+from dgl import AddSelfLoop
+from dgl.data import CiteseerGraphDataset, CoraGraphDataset, PubmedGraphDataset
+
+
+class SAGE(nn.Module):
+    def __init__(self, in_size, hid_size, out_size):
```
Comment (keli-wen) [inline examples/core/graphsage/node_classification.py:33]: `hid_size`-> `hidden_size`, please refer to [[Examples] Add Sampling Examples # 5777](https://github.com/dmlc/dgl/pull/5777) so that we can unify the specification.

File: examples/core/graphsage/node_classification.py [hunk 6]
```diff
@@ -0,0 +1,132 @@
...
+        # two-layer GraphSAGE-mean
+        self.layers.append(dglnn.SAGEConv(in_size, hid_size, "gcn"))
+        self.layers.append(dglnn.SAGEConv(hid_size, out_size, "gcn"))
+        self.dropout = nn.Dropout(0.5)
+
+    def forward(self, graph, x):
+        h = self.dropout(x)
+        for l, layer in enumerate(self.layers):
+            h = layer(graph, h)
+            if l != len(self.layers) - 1:
```
Comment (keli-wen) [inline examples/core/graphsage/node_classification.py:45]: Maybe you can use `is_last_layer = layer_idx == len (self.layers) - 1` to improve readability.

And I think we can make the naming of `h` and `l` more meaningful.

File: examples/core/graphsage/node_classification.py [hunk 7]
```diff
@@ -0,0 +1,132 @@
...
+        logits = model(g, features)
+        logits = logits[mask]
+        labels = labels[mask]
+        _, indices = torch.max(logits, dim=1)
+        correct = torch.sum(indices == labels)
+        return correct.item() * 1.0 / len(labels)
+
+
+def train(g, features, labels, masks, model):
+    # define train/val samples, loss function and optimizer
```
Comment (keli-wen) [inline examples/core/graphsage/node_classification.py:63]: Just to confirm, do we need to add `.` at each comment's end? @frozenbugs

File: examples/core/graphsage/node_classification.py (hunk 1 above)
Comment (caojy1998) [reply examples/core/graphsage/node_classification.py:38]: I check the doc for DGL. It seems that it is a typo here and I have corrected it.

File: examples/core/graphsage/node_classification.py (hunk 4 above)
Comment (caojy1998) [reply examples/core/graphsage/node_classification.py:132]: OK

File: examples/core/graphsage/node_classification.py (hunk 3 above)
Comment (caojy1998) [reply examples/core/graphsage/node_classification.py:48]: I will follow keli's idea to change this part and maintain consistency as well.

File: examples/core/graphsage/node_classification.py (hunk 7 above)
Comment (frozenbugs) [reply examples/core/graphsage/node_classification.py:63]: Yes, for all comment, as long as it is a sentence, let's follow the English gramma.

**Comment by frozenbugs** [comment]
Please also update the title / description of the PR

File: examples/core/graphsage/node_classification.py [hunk 8]
```diff
@@ -0,0 +1,131 @@
...
+    args = parser.parse_args()
+    print(f"Training with DGL built-in GraphSage module")
+
+    #####################################################################
+    # (HIGHLIGHT) This is a node classification task in which we read in
+    # the whole graph and train it on device. AddSelfLoop() will add
+    # self-loops for each of the node.
+    #####################################################################
+
+    # load and preprocess dataset
```
Comment (frozenbugs) [inline examples/core/graphsage/node_classification.py:100]:
```suggestion
    # Load and preprocess dataset.
```

File: examples/core/graphsage/node_classification.py [hunk 9]
```diff
@@ -0,0 +1,131 @@
...
+        help="Dataset name ('cora', 'citeseer', 'pubmed')",
+    )
+    args = parser.parse_args()
+    print(f"Training with DGL built-in GraphSage module")
+
+    #####################################################################
+    # (HIGHLIGHT) This is a node classification task in which we read in
+    # the whole graph and train it on device. AddSelfLoop() will add
+    # self-loops for each of the node.
+    #####################################################################
```
Comment (frozenbugs) [inline examples/core/graphsage/node_classification.py:98]: Is this a good highlight? This example shows to user how to leverage graph sage for node classification.

File: examples/core/graphsage/node_classification.py [hunk 10]
```diff
@@ -0,0 +1,131 @@
...
+        AddSelfLoop()
+    )  # by default, it will first remove self-loops to prevent duplication
+    if args.dataset == "cora":
+        data = CoraGraphDataset(transform=transform)
+    elif args.dataset == "citeseer":
+        data = CiteseerGraphDataset(transform=transform)
+    elif args.dataset == "pubmed":
+        data = PubmedGraphDataset(transform=transform)
+    else:
+        raise ValueError("Unknown dataset: {}".format(args.dataset))
```
Comment (keli-wen) [inline examples/core/graphsage/node_classification.py:111]: use fstring here~

File: examples/core/graphsage/node_classification.py (hunk 8 above)
Comment (keli-wen) [inline examples/core/graphsage/node_classification.py:100]: `Load and preprocess dataset.`

File: examples/core/graphsage/node_classification.py [hunk 11]
```diff
@@ -0,0 +1,131 @@
...
+    g = data[0]
+    device = torch.device("cuda" if torch.cuda.is_available() else "cpu")
+    g = g.int().to(device)
+    features = g.ndata["feat"]
+    labels = g.ndata["label"]
+    masks = g.ndata["train_mask"], g.ndata["val_mask"]
```
Comment (keli-wen) [inline examples/core/graphsage/node_classification.py:117]: Maybe we should explicitly use `()` to indicate tuple?

File: examples/core/graphsage/node_classification.py (hunk 10 above)
Comment (caojy1998) [reply examples/core/graphsage/node_classification.py:111]: OK

File: examples/core/graphsage/node_classification.py (hunk 8 above)
Comment (caojy1998) [reply examples/core/graphsage/node_classification.py:100]: Done

Comment (caojy1998) [reply examples/core/graphsage/node_classification.py:100]: Done

File: examples/core/graphsage/node_classification.py (hunk 11 above)
Comment (caojy1998) [reply examples/core/graphsage/node_classification.py]: Yeah, I asked Rui. It is better to add bracket for tuple.

**Review by frozenbugs — APPROVED** [review]

**Review by keli-wen — APPROVED** [review]

File: examples/pytorch/graphsage/train_full.py
[the remark this answers is not in the document]
Comment (caojy1998) [reply examples/pytorch/graphsage/train_full.py:1]: OK, I have revereted it.
